### ISSUE TITLE
[Caffe2] caffe2/python/caffe2_pybind11_state.so: undefined symbol: _ZN10onnx_torch11TensorProtoD1Ev

### ISSUE THREAD

**Issue by wang5566** [pytorch/pytorch#9519]
I built caffe2 from latest pytorch source code and sudo make install sucessfully.
But I when I run "python -c 'from caffe2.python import core'" from terminal, I got:

> WARNING:root:This caffe2 python run does not have GPU support. Will run in CPU only mode.
WARNING:root:Debug message: No module named caffe2_pybind11_state_hip
CRITICAL:root:Cannot load caffe2.python. Error: /home/workspace/pytorch/build/caffe2/python/caffe2_pybind11_state.so: undefined symbol: _ZN10onnx_torch11TensorProtoD1Ev

what's wrong?

**Comment by pjh5** [comment]
1. Did you run `git submodule update --init --recursive` right before you built Caffe2?
2. Did you install ONNX separately from source?
What were the commands you used to build Caffe2 and what was the https://caffe2.ai/docs/faq.html#what-is-the-cmake-output ?

**Comment by wang5566** [comment]
> Did you run git submodule update --init --recursive right before you built Caffe2?
> Did you install ONNX separately from source?
> What were the commands you used to build Caffe2 and what was the

I thought my problem is protobuf version. I reinstalled my system and built pytorch, torchision in anaconda from source, then I built caffe2 from source into system folder ,like "/usr/local" not using anaconda python, finally i built onnx from source instead of pip installation and pip install onnx-caffe2 tool to convert pytorch model to caffe2 model.
I solved it through this way.
